app/file_test.go

```diff
@@ -87,14 +87,21 @@ func TestDoUploadFile(t *testing.T) {
 }
 
 func TestUploadFile(t *testing.T) {
-	th := Setup(t)
+	th := Setup(t).InitBasic()
 	defer th.TearDown()
 
-	channelId := model.NewId()
+	channelId := th.BasicChannel.Id
 	filename := "test"
 	data := []byte("abcd")
 
-	info1, err := th.App.UploadFile(data, channelId, filename)
+	info1, err := th.App.UploadFile(data, "wrong", filename)
+	require.Error(t, err, "Wrong Channel ID.")
+	require.Nil(t, info1, "Channel ID does not exist.")
+
+	info1, err = th.App.UploadFile(data, "", filename)
+	require.Nil(t, err, "for legacy reasons it should accepts the empty channel id")
```

Comment:
Nit: no need to explicitly say "for legacy reasons", we can reword this as
```suggestion
	require.Nil(t, err, "empty channel IDs should be valid")
```